### PR TITLE
Optionally abort on unexpected missing commits; warn by default

### PR DESCRIPTION
**Pull request by stagnation**

### CODE REVIEW

File: git_toprepo.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
     List,
     Optional,
     Set,
+    TextIO,
```
Comment (stagnation) [inline git_toprepo.py:29]: remove TextIO

File: git_toprepo.py [hunk 2]
```diff
@@ -1509,13 +1519,14 @@ def get_commits_to_fetch() -> Set[CommitHash]:
...
-            print("ERROR: The referenced commits above could not be found")
+                print(f"{len(missing_commits)} unexpected missing commits.", file=sys.stderr)
+            print("Warning: The referenced commits above could not be found", file=sys.stderr)
             print(
                 "Either push the following commits or add them to the "
                 + "toprepo configuration."
             )
-            return None
+            if abort_on_missing:
+                sys.exit(1)
```
Comment (stagnation) [inline git_toprepo.py:1529]: Would serve us well to print a summary "Aborting due to previous warnings (--abort-on-unexpected-missing-commits)" before aborting.

Comment (moroten) [reply git_toprepo.py:1541]: Please keep `return None` as the rest of the code base aborts through the call chain and not using `sys.exit`.
```suggestion
                return None
```

File: git_toprepo.py [hunk 3]
```diff
@@ -1510,13 +1521,24 @@ def get_commits_to_fetch() -> Set[CommitHash]:
                 commit_hash_str = commit_hash.decode("utf-8")
                 print(f"\trev-{commit_hash_str} = {url}")
             if len(missing_commits) > max_missing_commits_to_print:
-                print(f"Totally {len(missing_commits)} unexpected missing commits.")
-            print("ERROR: The referenced commits above could not be found")
+                print(
+                    f"{len(missing_commits)} unexpected missing commits.",
```
Comment (ascallonisi) [inline git_toprepo.py:1525]: I'd clarify the message here with something like:
```suggestion
                    f"and {len(missing_commits) - max_missing_commits_to_print} more unexpected missing commits.",
```

**Review by ascallonisi — CHANGES_REQUESTED** [review]

File: git_toprepo.py (hunk 2 above)
Comment (stagnation) [reply git_toprepo.py:1541]: Okay, if you prefer. Though if we treat it as a library just returning `None` is not a very clear signal that the caller should exit. And if the caller does not exit, then the error print that says "aborting" is misplaced. I have updated the return, but we should keep this in mind, that `None` is not customarily a clear signal to exit everything. Hmm.

File: git_toprepo.py [hunk 4]
```diff
@@ -2605,6 +2637,12 @@ def _parse_arguments(argv: List[str]):
         dest="do_filter",
         help="Fetch from the remote but skip monorepo filtering.",
     )
+    fetch_parser.add_argument(
```
[the remark this answers is not in the document]
Comment (stagnation) [reply git_toprepo.py]: Yepp! Good catch, thank you.

**Review by ascallonisi — APPROVED** [review]

**Pull request merged**